floris/tools/optimization/scipy/yaw.py

```diff
@@ -294,40 +371,47 @@ def reinitialize_opt(
                 {'std_wd': 4.95, 'std_yaw': 1.75, 'pmf_res': 1.0,
                 'pdf_cutoff': 0.995} are used. Defaults to None.
+            turbine_weights (iterable, optional): weighing terms that allows the
+                user to emphasize power gains at particular turbines or
+                completely ignore power gains from other turbines. The array
+                of turbine powers from floris is multiplied with this array in
+                the calculation of the objective function. Defaults to an array
+                with all values 1.0 and length equal to the number of turbines.
             calc_init_power (bool, optional): If True, calculates initial wind
                 farm power for each set of wind conditions. Defaults to True.
         """
         if minimum_yaw_angle is not None:
             self.minimum_yaw_angle = minimum_yaw_angle
         if maximum_yaw_angle is not None:
             self.maximum_yaw_angle = maximum_yaw_angle
+        self.x_baseline = [
+                turbine.yaw_angle
+                for turbine in self.fi.floris.farm.turbine_map.turbines
+        ]
         if x0 is not None:
             self.x0 = x0
+            if not all(np.array(x0) == np.array(self.x_baseline)):
+                print("WARNING: Baseline yaw angle in FLORIS differ from initial optimization conditions (self.x0)")
         else:
-            self.x0 = [
-                turbine.yaw_angle
-                for turbine in self.fi.floris.farm.turbine_map.turbines
-            ]
-        self.x0_norm = self._norm(
-            np.array(self.x0), self.minimum_yaw_angle, self.maximum_yaw_angle
-        )
+            self.x0 = self.x_baseline
+        self.bnds = bnds
         if bnds is not None:
-            self.bnds = bnds
             self.minimum_yaw_angle = np.min([bnds[i][0] for i in range(self.nturbs)])
             self.maximum_yaw_angle = np.max([bnds[i][1] for i in range(self.nturbs)])
         else:
             self._set_opt_bounds(self.minimum_yaw_angle, self.maximum_yaw_angle)
-        self.bnds_norm = [
-            (
-                self._norm(
-                    self.bnds[i][0], self.minimum_yaw_angle, self.maximum_yaw_angle
-                ),
-                self._norm(
-                    self.bnds[i][1], self.minimum_yaw_angle, self.maximum_yaw_angle
-                ),
-            )
-            for i in range(self.nturbs)
-        ]
+
+        # Check if x0 and x_baseline exceed any bounds
+        for ti in range(self.nturbs):
+            if self.x_baseline[ti] < self.bnds[ti][0]:
+                print("WARNING: Baseline yaw angle in FLORIS for turbine %d is %.3f and exceeds lower bound constraint of %.3f." % (ti, self.x_baseline[ti], self.bnds[ti][0]))
+            if self.x_baseline[ti] > self.bnds[ti][1]:
+                print("WARNING: Baseline yaw angle in FLORIS for turbine %d is %.3f and exceeds upper bound constraint of %.3f." % (ti, self.x_baseline[ti], self.bnds[ti][1]))
```

Comment:
Also don't think the first two warnings need to be there since the bounds are supposed to be the constraints for the optimal values, not necessarily the baseline values. But also can't think of many cases where the bounds would be set that way, so either way should be ok.